### ISSUE TITLE
Asan and "AddressSanitizer: SEGV on unknown address 0xfffffef0" in VMAC

### ISSUE THREAD

**Issue by noloader** [weidai11/cryptopp#304]
`cryptest.sh` reports the following using g++ 4.9.2-10 on Debian 8.6/i686. It occurs with and without `-std=c++03`. Its also duplicates under g++5.3 on Fedora 23/i686. Finally, it duplicates with `vmac.cpp` from Crypto++ 5.6.2.

```
make CXXFLAGS="-DDEBUG -g3 -O0 -std=c++03 -march=native -pipe -fsanitize=address -fno-omit-frame-pointer -D_GLIBCXX_DEBUG"
...

./cryptest.exe v
...

Testing MAC algorithm VMAC(AES)-64.
.ASAN:SIGSEGV
=================================================================
==27062==ERROR: AddressSanitizer: SEGV on unknown address 0xfffffef0 (pc 0x08355529 sp 0xbf9830f0 bp 0xbf9832c8 T0)
    #0 0x8355528 in CryptoPP::VMAC_Base::VHASH_Update_SSE2(unsigned long long const*, unsigned int, int) /home/jwalton/cryptopp/vmac.cpp:477
    #1 0x8357094 in CryptoPP::VMAC_Base::VHASH_Update(unsigned long long const*, unsigned int) /home/jwalton/cryptopp/vmac.cpp:811
    #2 0x8356289 in CryptoPP::VMAC_Base::TruncatedFinal(unsigned char*, unsigned int) /home/jwalton/cryptopp/vmac.cpp:887
    #3 0x824a95c in CryptoPP::HashTransformation::TruncatedVerify(unsigned char const*, unsigned int) /home/jwalton/cryptopp/cryptlib.cpp:412
    #4 0x856705b in CryptoPP::HashVerificationFilter::LastPut(unsigned char const*, unsigned int) /home/jwalton/cryptopp/filters.cpp:840
    #5 0x8560503 in CryptoPP::FilterWithBufferedInput::PutMaybeModifiable(unsigned char*, unsigned int, int, bool, bool) /home/jwalton/cryptopp/filters.cpp:437
    #6 0x806278f in CryptoPP::FilterWithBufferedInput::Put2(unsigned char const*, unsigned int, int, bool) /home/jwalton/cryptopp/filters.h:339
    #7 0x805c21e in CryptoPP::BufferedTransformation::MessageEnd(int, bool) /home/jwalton/cryptopp/cryptlib.h:1444
    #8 0x81da800 in TestDigestOrMAC(std::__debug::map<std::string, std::string, std::less<std::string>, std::allocator<std::pair<std::string const, std::string> > >&, bool) /home/jwalton/cryptopp/datatest.cpp:604
    #9 0x81dc982 in TestDataFile(std::string, CryptoPP::NameValuePairs const&, unsigned int&, unsigned int&) /home/jwalton/cryptopp/datatest.cpp:804
    #10 0x81dd117 in RunTestDataFile(char const*, CryptoPP::NameValuePairs const&, bool) /home/jwalton/cryptopp/datatest.cpp:843
    #11 0x813d0d9 in ValidateVMAC() /home/jwalton/cryptopp/validat1.cpp:3039
    #12 0x8125119 in ValidateAll(bool) /home/jwalton/cryptopp/validat1.cpp:141
    #13 0x80588c0 in Validate(int, bool, char const*) /home/jwalton/cryptopp/test.cpp:894
    #14 0x804ffdd in main /home/jwalton/cryptopp/test.cpp:360
    #15 0xb6f4aa62 in __libc_start_main (/lib/i386-linux-gnu/i686/cmov/libc.so.6+0x19a62)
    #16 0x804d7e0 (/home/jwalton/cryptopp/cryptest.exe+0x804d7e0)

AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV /home/jwalton/cryptopp/vmac.cpp:477 CryptoPP::VMAC_Base::VHASH_Update_SSE2(unsigned long long const*, unsigned int, int)
==27062==ABORTING
```

-----

In contrast, Valgrind reports no problems:

```
$ valgrind ./cryptest.exe tv vmac
==9343== Memcheck, a memory error detector
==9343== Copyright (C) 2002-2015, and GNU GPL'd, by Julian Seward et al.
==9343== Using Valgrind-3.12.0.SVN and LibVEX; rerun with -h for copyright info
==9343== Command: ./cryptest.exe tv vmac
==9343== 
Using seed: 1474675008      

Testing MAC algorithm VMAC(AES)-64.
...........
Testing MAC algorithm VMAC(AES)-128.
...........
Tests complete. Total tests = 22. Failed tests = 0.
==9343== 
==9343== HEAP SUMMARY:
==9343==     in use at exit: 18,944 bytes in 1 blocks
==9343==   total heap usage: 2,997 allocs, 2,996 frees, 6,197,441 bytes allocated
==9343== 
==9343== LEAK SUMMARY:
==9343==    definitely lost: 0 bytes in 0 blocks
==9343==    indirectly lost: 0 bytes in 0 blocks
==9343==      possibly lost: 0 bytes in 0 blocks
==9343==    still reachable: 18,944 bytes in 1 blocks
==9343==         suppressed: 0 bytes in 0 blocks
==9343== Rerun with --leak-check=full to see details of leaked memory
==9343== 
==9343== For counts of detected and suppressed errors, rerun with: -v
==9343== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
```

**Comment by noloader** [comment]
X86 ASM was disabled until we can sort this out. Also see [Commit 0a22957481b7a9b](http://github.com/weidai11/cryptopp/commit/0a22957481b7a9bd963bec94bb81cc6df6f0b413).

**Comment by noloader** [comment]
X86 ASM was re-enabled at [Commit 51884539cbbe1783](https://github.com/weidai11/cryptopp/commit/51884539cbbe17834e4a3c8436b606c211668fdf).

**Comment by noloader** [comment]
Finally fixed at [Commit f7986129dfe3](https://github.com/weidai11/cryptopp/commit/f7986129dfe3).
